pkg/target/target.go

```diff
@@ -47,12 +48,22 @@ const wildcardNSPattern = `^(\*|\*-)?[a-z0-9]([-a-z0-9]*[a-z0-9])?$|^[a-z0-9]([-
 
 var _ handler.TargetHandler = &K8sValidationTarget{}
 
-type K8sValidationTarget struct{}
+type K8sValidationTarget struct {
+	cache nsCache
+}
 
 func (h *K8sValidationTarget) GetName() string {
 	return "admission.k8s.gatekeeper.sh"
 }
 
+func (h *K8sValidationTarget) Add(key string, object interface{}) error {
+	return h.cache.Add(key, object)
+}
+
+func (h *K8sValidationTarget) Remove(key string) {
```

Comment:
Make sure to test that removal works.

Reply:
While not "clean" (since it would make the tests serve multiple purposes), you could add the removal test to automatically happen at the end of all of your add tests. One way of doing this would be:

```
tests := []struct{
  ...
  removeNs map[string]*Namespace
}

...

for key, ns := range tt.addNs {
  h.Add(key, ns)
}

for key, ns := range tt.removeNs {
  h.Remove(key)
}
```
and then validate the state of the cache.

Reply:
I added a "wantNs" struct to the tests, which can signify if the wanted ns should exist.